hbase-client/src/main/java/org/apache/hadoop/hbase/shaded/protobuf/ProtobufUtil.java

```diff
@@ -2293,6 +2293,13 @@ public static TableName toTableName(HBaseProtos.TableName tableNamePB) {
         .setQualifier(UnsafeByteOperations.unsafeWrap(tableName.getQualifier())).build();
   }
 
+  public static List<HBaseProtos.TableName> toProtoTableNameList(List<TableName> tableNameList) {
+    if (tableNameList == null) {
+      return new ArrayList<>();
```

Comment:
nit: `Collections.emptyList()` ?

Reply:
`emptyList` returns a non-modifiable collection. The other methods here follow this pattern of returning a modifiable list. I haven't evaluated all the call-sites to see if `emptyList` would be acceptable. Even if not, probably these `ArrayLists` should be replaced with `LinkedList`.

Reply:
For the purpose of this PR, if we return non-modifiable collection, `getTableDescriptors` in HMaster will interpret it as expected: `if (tableNameList == null || tableNameList.isEmpty()) {`, however since this is generic method and we are following the same approach for majority of util methods on this class, we can keep it same way.

> Even if not, probably these ArrayLists should be replaced with LinkedList.

You meant as separate task right? +1

Reply:
Yes, I mean as a general cleanup/improvement to the methods in this class, separate from this issue.